### ISSUE TITLE
Add a static Boot App to host and serve OOTB Apps

### ISSUE THREAD

**Issue by sabbyanandan** [spring-cloud/spring-cloud-dataflow-samples#72]
As a developer, I'd like to add a static Boot App as a placeholder template for OOTB Apps, so the users can list the desired App coordinates to download, bundle, and serve the uber-jar's. This App would be an alternative approach to Maven artifact resolution, and it would also be useful for environments with no-internet connectivity.

**Notes:**
A potential directory structure could be:

```
(at root)
./download.sh
./stream-apps.properties
	time-source-1.0.0.RELEASE
	log-sink-1.0.0.RELEASE
./task-apps.properties
	timestamp-2.0.0.RELEASE
```

1) Executing `download.sh` would populate:

```
./src/main/resources/static/stream-apps/time-source-1.0.0.RELEASE.jar
./src/main/resources/static/stream-apps/log-sink-1.0.0.RELEASE.jar
./src/main/resources/static/task-apps/timestamp-2.0.0.RELEASE.jar
```

Basically, `stream-apps.properties` and `stream-apps.properties` serve as a template; users can add app coordinates in it, the download script will `wget` it from the Spring repo.

2) Executing `mvn package` would build the static App with all the JARs in it. 

3) This static App can now serve the Apps via `localhost` or on a CF route in PCF _(i.e. `cf push` ready)_ etc.

**Comment by dturanski** [comment]
Initial implementation in https://github.com/dturanski/scdf-app-tool.  Per @markpollack - we don't want to publish binary distributions to a git repo.  The dist is currently packaged as currently `*.zip, *.tar.bz2, *.tar.gz`, each containing the Spring Shell tool to import and manage the repo contents, the task and stream properties files and the source for the Spring Boot application that will serve the static resources.   There is a precedent (e.g., Spring XD) for publishing these archives to `repo.spring.io`.  I'm fine with that.  The current maven coordinates are 

```
<groupId>org.springframework.cloud</groupId>
<artifactId>scdf-app-tool</artifactId>
<version>1.0.0.BUILD-SNAPSHOT</version>
```

Maybe we can come up with a better name than `scdf-app-tool`.  The tool is what a developer uses to build the `scdf-app-repo`.  IMHO, this is specific to SCDF. Even though the apps are not, the use case is to register apps in SCDF in situations where the environment does not support access to an external Maven repo.  Since we are publishing to Maven the artifactId is important.  The pom needs to be updated to publish the binaries, so we should agree on the name first.

**Comment by dturanski** [comment]
Also, this doesn't fit into any of the existing git repos, so we should create a new one which will likely be the same as the artifactId or something like `spring-cloud-dataflow-tools`

**Comment by sabbyanandan** [comment]
This is addressed via: https://github.com/spring-cloud-stream-app-starters/scdf-app-repo
